### PR TITLE
Add adr for on call process.

### LINKED ISSUE #4: Create a process for SRE On-Call and support handling for Operate First Cloud
Goal: 

Spread the work of being on-call across the SRE team.

Acceptance Criteria: 

- [ ] Documentation on a set of processes for the Operate First Community Cloud SRE team
- [ ] The person on-call should be responsible for tackling most of the SRE work, and support queries, during their period of on call. 
- [ ] On-Call role also involves resolving existing issues
- [ ] On-Call role also involves updating existing docs, and adding new resolution/support documentation as they perform their on-call duties. Such that future problems are able to be self serviced without involvement of SRE team (if applicable).
- [ ] Process should encourage breaking down silos

### PR DESCRIPTION
**Pull request by HumairAK**
Resolve: https://github.com/open-services-group/scrum/issues/4

### CODE REVIEW

File: adr/0021-sre-cloud-support.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+* Status: accepted
+* Deciders: durandom
+* Date: 2022-07-25
+
+Technical Story: https://github.com/open-services-group/scrum/issues/4
+
+## Context and Problem Statement
+
+In general, the individual or team that onboards a service often assumes the sole role of operating that service
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
In general, the individual or team that onboards a service often assumes the sole role of operating that service indefinitely. There are various issues associated with this:
```

File: adr/0021-sre-cloud-support.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+indefinitely. There are various issues associated with this:
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md (hunk 1 above)
Comment (schwesig) [reply adr/0021-sre-cloud-support.md]: following the idea of: every sentence on one line for easy suggestions

File: adr/0021-sre-cloud-support.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* Other members of the SRE team are unable to offer sufficient support for the service, support here can refer to:
+  * Operational support (improvements, incidents, fixes, patches, upgrades, etc.)
+  * Responding to end user questions via GitHub, Slack, Mailing List pertaining to the service
+* Siloed knowledge results in fewer team members being able to perform PR reviews pertaining to the service
+* Documentation for the app/service suffers, since now only one person is able to write/understand them
+* The one (or few) engineers responsible for maintaining the service have to devote large portions of their engineer
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* The one (or few) engineers responsible for maintaining the service have to devote large portions of their engineer time to operate the service, as opposed to building out new features in Operate First
```

File: adr/0021-sre-cloud-support.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  time to operate the service, as opposed to building out new features in Operate First.
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md (hunk 3 above)
Comment (schwesig) [reply adr/0021-sre-cloud-support.md]: removing also the period "." to align the layout (some list items have periods, some don't)

File: adr/0021-sre-cloud-support.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+The proposed solution is to apply an SRE On Call procedure in such a way as to alleviate the aforementioned concerns.
+
+## Decision Drivers
+
+* All SRE members in Operate First should have the opportunity to be On Call rotation
+* On Call rotation should be able to scale with work and not take up large portions of an engineer's sprint
+* On Call person should have sufficient support to perform on call duties
+* Not aiming for 24/7 on call coverage
+* Need to maintain and improve documentation used for on call duties.
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* Need to maintain and improve documentation used for on call duties
```

Comment (schwesig) [reply adr/0021-sre-cloud-support.md]: removing also the period "." to align the layout (some list items have periods, some don't)

File: adr/0021-sre-cloud-support.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+## Considered Options
+
+The only options to consider are the duration each engineer spends on call, i.e. the frequency of rotations:
+
+1) On call rotations happen every week
+2) On call rotations happen every day
+
+## Decision Outcome
+
+Selected option 2) as option 1) could result in a major portion of an engineer's sprint time being spent on call and
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
Selected option 2) as option 1) could result in a major portion of an engineer's sprint time being spent on call and thus does not scale well.
```

File: adr/0021-sre-cloud-support.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+thus does not scale well.
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+## Process
+
+### On Call List
+
+Maintain an On Call list of engineers to perform site reliability in the Operate First Cloud. Official list can be
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
Maintain an On Call list of engineers to perform site reliability in the Operate First Cloud. Official list can be maintained [here][on-call-list].
```

File: adr/0021-sre-cloud-support.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+maintained [here][on-call-list].
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+### Schedule
+
+Scheduled rotations happen every working day. When an engineer is on call, they are expected to perform the following
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
Scheduled rotations happen every working day. When an engineer is on call, they are expected to perform the following on call duties:
```

File: adr/0021-sre-cloud-support.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+on call duties:
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+### On Call Duties
+
+When on call, perform the following duties:
+
+* Respond to all end-user questions/comments/concerns regarding operations in:
+  * Operate First Slack #support, #operations, #general
+  * Operate First Mailing list
+  * Operate First Operations Repos (i.e, `apps`, `operations`)
+* Be first responder to any issues, incidents, and other operational issues for services managed by SRE team
+* Follow documentations to resolve any issues, if no documentation and/or runbook exists, ensure they are updated once
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* Follow documentations to resolve any issues, if no documentation and/or runbook exists, ensure they are updated once the issues are resolved
```

File: adr/0021-sre-cloud-support.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+  the issues are resolved
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+* If the on call engineer does not know how to resolve an issue, and no documentation exist, the onus is on the on call
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* If the on call engineer does not know how to resolve an issue, and no documentation exist, the onus is on the on call engineer to reach out to the rest of the SRE team for assistance
```

File: adr/0021-sre-cloud-support.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+  engineer to reach out to the rest of the SRE team for assistance.
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

File: adr/0021-sre-cloud-support.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+* In the event on call engineer is assisted by another engineer, it is still the on call engineer's responsibility to
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* In the event on call engineer is assisted by another engineer, it is still the on call engineer's responsibility to ensure the resolution steps are fully documented in the form of a runbook, for the next on call engineer
```

File: adr/0021-sre-cloud-support.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+  ensure the resolution steps are fully documented in the form of a runbook, for the next on call engineer.
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
```

**Review by schwesig — COMMENTED** [review]
Do we have one or more links to the go-to run books, or where/ how to add/ update the documentation?

**Comment by schwesig** [comment]
@VannTen
@mimotej 
@SamoKopecky 
@tumido 
@Gregory-Pereira

**Comment by HumairAK** [comment]
> Do we have one or more links to the go-to run books, or where/ how to add/ update the documentation?

@schwesig https://www.operate-first.cloud/apps/CONTRIBUTING.html

**Comment by schwesig** [comment]
> > Do we have one or more links to the go-to run books, or where/ how to add/ update the documentation?
> 
> @schwesig [operate-first.cloud/apps/CONTRIBUTING.html](https://www.operate-first.cloud/apps/CONTRIBUTING.html)

Should we add this link to the document? Or is this common knowledge/ already exists in readme etc and I missed that?

File: adr/0021-sre-cloud-support.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+
+The proposed solution is to apply an SRE On Call procedure in such a way as to alleviate the aforementioned concerns.
+
+## Decision Drivers
+
+* All SRE members in Operate First should have the opportunity to be On Call rotation
+* On Call rotation should be able to scale with work and not take up large portions of an engineer's sprint
+* On Call person should have sufficient support to perform on call duties
+* Not aiming for 24/7 on call coverage
+* Need to maintain and improve documentation used for on call duties
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md]:
```suggestion
* Need to maintain and improve [documentation](https://www.operate-first.cloud/apps/CONTRIBUTING.html) used for on call duties
```

File: adr/0021-sre-cloud-support.md [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+### On Call Duties
+
+When on call, perform the following duties:
+
+* Respond to all end-user questions/comments/concerns regarding operations in:
+  * Operate First Slack #support, #operations, #general
+  * Operate First Mailing list
+  * Operate First Operations Repos (i.e, `apps`, `operations`)
+* Be first responder to any issues, incidents, and other operational issues for services managed by SRE team
+* Follow documentations to resolve any issues, if no documentation and/or runbook exists, ensure they are updated once the issues are resolved
```
Comment (schwesig) [inline adr/0021-sre-cloud-support.md:60]:
```suggestion
* Follow [documentations](https://www.operate-first.cloud/apps/CONTRIBUTING.html) to resolve any issues, if no documentation and/or runbook exists, ensure they are updated once the issues are resolved
```

Comment (schwesig) [reply adr/0021-sre-cloud-support.md:60]: maybe here?

File: adr/0021-sre-cloud-support.md (hunk 18 above)
Comment (schwesig) [reply adr/0021-sre-cloud-support.md]: maybe here?

**Review by schwesig — COMMENTED** [review]
/lgtm
the idea about adding the link to the docs https://www.operate-first.cloud/apps/CONTRIBUTING.html is not a blocker for me.

**Review by schwesig — COMMENTED** [review]
/lgtm
adding https://www.operate-first.cloud/apps/CONTRIBUTING.html is not a blocker

File: adr/0021-sre-cloud-support.md (hunk 19 above)
Comment (HumairAK) [reply adr/0021-sre-cloud-support.md:60]: added!

**Comment by durandom** [comment]
good job @HumairAK  and @schwesig

**Pull request merged**